### PR TITLE
Debug logging changed (closes #1289)

### LINKED ISSUE #1289: Change debugging logging style
### Are you requesting a feature or reporting a bug?

feature 

### What is the current behavior?

debugging messages are added one by one in console (for each debugging stop point)

### What is the expected behavior?

We should show one debugging message and update it on each debugging stops

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AlexanderMoskovkin 
Now debug logging looks like this:
debug-log
Information about debugger stop points appears always in the end of log. 
And debugger message for each browser is updatable
updatable
Result of test run is the same as before, because we will remove all debugging messages after the end of debugging .

### CODE REVIEW

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

File: src/test-run/index.js [hunk 1]
```diff
@@ -438,7 +440,7 @@ ServiceMessages[CLIENT_MESSAGES.waitForFileDownload] = function (msg) {
 ServiceMessages[CLIENT_MESSAGES.showDebuggerMessage] = function (msg) {
     this.debugLog.driverMessage(msg);
 
-    showDebuggerMessage(this.currentDriverTask.callsite, this.browserConnection.userAgent);
+    debuggerMessage.show(this.id, this.browserConnection.userAgent, this.currentDriverTask.callsite);
```
Comment (inikulin) [inline src/test-run/index.js]: It should be named `debugLogger`.
`show` -> `showBreakpoint`,
`remove ` -> `hideBreakpoint`
`id` -> `testRunId`

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
@testcafe-build-bot /retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot /retest

**Comment by helen-dikareva** [comment]
FPR

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

File: src/notifications/debug-logger.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    inLogging: false,
+
+    streamsOverridden: false,
+
+    _overrideStream (stream) {
+        var initialWrite = stream.write;
+
+        stream.write = function () {
+            if (!this.inLogging) {
```
Comment (inikulin) [inline src/notifications/debug-logger.js]: `inLogging` - bad name

File: src/notifications/debug-logger.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                this.inLogging = true;
+
+                logUpdate.clear();
+                logUpdate.done();
+
+                initialWrite.apply(stream, arguments);
+
```
Comment (inikulin) [inline src/notifications/debug-logger.js]: Who restores debug log here?

File: src/notifications/debug-logger.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    },
+
+    _overrideStreams () {
+        this._overrideStream(process.stdout);
+        this._overrideStream(process.stderr);
+
+        this.streamsOverridden = true;
+    },
+
+    _getMessageIndex (testRunId) {
```
Comment (inikulin) [inline src/notifications/debug-logger.js]: We have lodash `find` for this

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

File: src/notifications/debug-logger.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+export default {
+    messages: [],
+
+    debugLogging: false,
+
+    streamsOverridden: false,
+
+    _overrideStream (stream) {
+        var initialWrite = stream.write;
+        var that         = this;
```
Comment (inikulin) [inline src/notifications/debug-logger.js]: Please, don't.

File: src/notifications/debug-logger.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+        // NOTE: we cannot use arrow function here because 'this' should
+        // be a 'stream' object otherwise stream.write will raise error
+        stream.write = function () {
```
Comment (inikulin) [inline src/notifications/debug-logger.js]: Just use arrow function

File: src/notifications/debug-logger.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            if (that.debugLogging) {
+                initialWrite.apply(stream, arguments);
+                return;
```
Comment (inikulin) [inline src/notifications/debug-logger.js]: Please, try to use consistent returns. Do `if/else` here

**Comment by helen-dikareva** [comment]
Please do not merge!

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR
/cc @AndreyBelym

**Pull request merged**